File: game_plugins/sixpack/src/sc/plugin2014/moves/DebugHint.java
package sc.plugin2014.moves;

import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

/**
 * Ein Debughinweis ist ein Container für einen String der einem Zug
 * beigefuegt werden kann. Beigefuegte Debughints werden direkt in der
 * grafischen Oberflaeche des Plugins angezeigt, wenn die Debugansicht gewaehlt
 * wurde. <br>
 * <br>
 * Dies ermoeglich das schnellere Debuggen von Clients und besseres
 * Konfigurieren von Strategien, denn es muessen keine Konsolenausgaben gesucht
 * werden und die Hinweise werden immer zum passenden Zug angezeigt.
 * 
 * @author tkra
 * 
 */
public class DebugHint implements Cloneable {

    @XStreamAsAttribute
    private final String content;

    public DebugHint() {
        content = "";
    }

    public DebugHint(String key, String value) {

        key = key == null ? "" : key;
        value = value == null ? "" : value;

        if (!(key.equals("") && value.equals(""))) {
            content = key + " = " + value;
        }
        else {
            content = key + value;
        }

    }

    public DebugHint(final String content) {
        this.content = content == null ? "" : content;
    }

    public String getContent() {
        return content;
    }

    @Override
    protected Object clone() throws CloneNotSupportedException {
        return new DebugHint(getContent());
    }

}

File: socha-sdk/src/framework/sc/shared/DebugHint.kt
package sc.shared

import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/**
 * An additional note to be appended to a move.
 * These notes can then be displayed in the GUI if debugging is enabled.
 *
 * This enables faster debugging of clients and easier configuring of strategies as
 * important information gets displayed together with the corresponding move (instead of the CLI)
 * ===
 * Ein Debughinweis ist ein Container für einen String, der einem Zug beigefuegt
 * werden kann. Beigefuegte Debughints werden direkt in der grafischen
 * Oberflaeche des Plugins angezeigt, wenn die Debugansicht gewaehlt wurde.
 *
 * Dies ermöglicht das schnellere Debuggen von Clients und besseres
 * Konfigurieren von Strategien, denn es müssen keine Konsolenausgaben gesucht
 * werden und die Hinweise werden immer zum passenden Zug angezeigt.
 */
class DebugHint(@XStreamAsAttribute val content: String = "") : Cloneable {
    /**
     * This constructs a hint of the form "key = value"
     */
    constructor(key: String, value: String) : this("$key = $value")
}

